### ISSUE TITLE
Payment gateways don't register if any payment gateway does not register

### ISSUE THREAD

**Issue by JeffPyeBrook** [wp-e-commerce/WP-e-Commerce#2047]
The register_dir method returns as soon as any payment gateway has an error rather than trying to register all of the gateways. 

This means that if one gateway, appearing early in the list, has unique requirements like requiring mcrypt or curl or theme engine 2, and those requirements are not met, all remaining gateways fail to register.

Does not seem like the correct behavior?

**Comment by JeffPyeBrook** [comment]
Below is the code fragment in question, perhaps the return should log the error instead?

```
		foreach ( $files as $file ) {
			$path = $dir . $file;

			if ( pathinfo( $path, PATHINFO_EXTENSION ) != 'php' || in_array( $file, array( '.', '..' ) ) || is_dir( $path ) ) {
				continue;
			}

			$return = self::register_file( $path );

			if ( is_wp_error( $return ) ) {
				return $return;
			}
		}
```

**Comment by JustinSainton** [comment]
Resolved in b75b0950909dc793f83fe681e76c97e051aadce0
